src/main/httpCommandServer.ts

```diff
@@ -0,0 +1,103 @@
+import fs from 'fs';
+import http from 'http';
+import path from 'path';
+
+import Logging from '@/utils/logging';
+import paths from '@/utils/paths';
+
+type ServerState = {
+  user: string;
+  password: string;
+  port: number;
+  pid: number;
+}
+
+const console = Logging.server;
+const ServerPort = 6107;
+const ServerUsername = 'user';
+const ServerFileBasename = 'rd-engine.json';
+
+export default class HttpCommandServer {
+  protected server = http.createServer();
+  protected password = randomStr();
+  protected stateInfo: ServerState = {
+    user:     ServerUsername,
+    password: this.password,
+    port:     ServerPort,
+    pid:      process.pid,
+  };
+
+  async init() {
+    const statePath = path.join(paths.appHome, ServerFileBasename);
+
+    await fs.promises.writeFile(statePath,
+      JSON.stringify(this.stateInfo, undefined, 2),
+      { mode: 0o600 });
+    this.server.listen(ServerPort, '127.0.0.1');
+    console.log(`Listening on port ${ ServerPort }, user: ${ ServerUsername },  password: ${ this.password }`);
+    this.server.on('request', this.handleRequest.bind(this));
+    this.server.on('error', (err) => {
+      console.log(`Error: ${ err }`);
+    });
+  }
+
+  shutdown() {
+    this.server.close();
+  }
+
+  protected handleRequest(request: http.IncomingMessage, response: http.ServerResponse) {
+    if (!this.basicAuth(request.headers.authorization ?? '')) {
+      response.writeHead(401, { 'Content-Type': 'text/plain' });
+    } else {
+      response.writeHead(200, { 'Content-Type': 'text/plain' });
+      response.write('Nothing to see here yet.');
+    }
+    response.end();
```

Comment:
Should this be in `try`/`finally`? I'm not familiar with the API conventions here…